ci/lava/tests/test-boot-component-update.py

```diff
@@ -0,0 +1,164 @@
+)
+
+
+class TestBootComponentUpdate:
+    """Class to encapsulate the testing of core components on a DUT."""
+
+    update_payload = None
+
+    def test_setup(self, update_payload):
+        """Setup test."""
+        TestBootComponentUpdate.update_payload = update_payload
+
+    def test_get_current_boot_component_data(
+        self, dut_addr, execute_helper, update_component_name
+    ):
+        """Save data of the component we are about to update.
+
+        For bootloader1 and bootloader2, it saves the whole bootloader
+        partition onto a file. This will be then used to compare its content
+        after the update.
+
+        For the kernel update it's enough to get the output of "uname -a"
+        """
+        if update_component_name == "kernel":
+            exit_code, output = get_kernel_version(dut_addr, execute_helper)
+            assert exit_code == 0
+            TestBootComponentUpdate.kernel_version_pre = output
+        elif "wks_bootloader" in update_component_name:
+            exit_code, component_partition_file = dd_bootloader_component(
+                dut_addr,
+                execute_helper,
+                update_component_name,
+                "partition",
+                "pre",
+            )
+            assert exit_code == 0
+            TestBootComponentUpdate.component_partition_file_pre = (
+                component_partition_file
+            )
+        else:
+            raise AssertionError(
+                "Update of component {} not supported".format(
+                    update_component_name
+                )
+            )
+
+    @pytest.mark.pelion
+    def test_component_update_pelion(
+        self, dut_addr, execute_helper, update_component_name
+    ):
+        """Test update component using Pelion."""
+        pass
+
+    @pytest.mark.mbl_cli
+    def test_component_update_mbl_cli(
+        self, dut_addr, execute_helper, update_component_name
+    ):
+        """Test update component using mbl-cli."""
+        # Copy the payload onto the DUT
+        exit_code, output, error = execute_helper.send_mbl_cli_command(
+            ["put", TestBootComponentUpdate.update_payload, "/scratch/"],
+            dut_addr,
+        )
+        assert exit_code == 0
+
+        # Run the update
+        payload_name = os.path.basename(TestBootComponentUpdate.update_payload)
+        cmd = "mbl-firmware-update-manager --assume-yes /scratch/{}".format(
+            payload_name
+        )
+        exit_code, output, error = execute_helper.send_mbl_cli_command(
+            ["shell", 'sh -l -c "{}"'.format(cmd)], dut_addr
+        )
+        assert "Content of update package installed" in output
+
+    @pytest.mark.mbl_cli
+    def test_dut_online_after_reboot(
+        self, dut, execute_helper, update_component_name
+    ):
+        """Wait the DUT to be back online after the reboot."""
+        # Wait some time before discovering again the DUT
+        time.sleep(10)
+        while True:
```

Comment:
We should probably put the `dut_address` check in the `while` statement e.g `while not dut_address`